xarray/core/variable.py

```diff
@@ -185,7 +185,7 @@ def _as_array_or_item(data):
 
     TODO: remove this (replace with np.asarray) once these issues are fixed
     """
-    data = np.asarray(data)
+    data = np.asanyarray(data)
```

Comment:
This method should probably be renamed, e.g., `_as_any_array_or_item`, just to make it super clear what it's doing (also update the TODO in the docstring)